ledger/block/src/transactions/confirmed/mod.rs

```diff
@@ -142,6 +142,15 @@ impl<N: Network> ConfirmedTransaction<N> {
         }
     }
 
+    pub fn variant(&self) -> String {
+        match self {
+            Self::AcceptedDeploy(..) => "accepted deploy".to_string(),
+            Self::AcceptedExecute(..) => "accepted execute".to_string(),
+            Self::RejectedDeploy(..) => "rejected deploy".to_string(),
+            Self::RejectedExecute(..) => "rejected execute".to_string(),
+        }
+    }
```

Comment:
We can save 1 allocation by changing the return type to `&str` and avoid calling `.to_string()`.

Also, I realized there are many functions here (including this one) that can be changed from `pub fn` to `pub const fn`.